xbmc/cores/VideoPlayer/DVDDemuxers/DemuxTimeline.cpp

```diff
@@ -0,0 +1,297 @@
+{
+  --chapterIdx;
+  if (chapterIdx < 0 || unsigned(chapterIdx) >= m_chapters.size())
+    return 0;
+  return (m_chapters[chapterIdx].startDispTime + 999) / 1000;
+}
+
+void CDemuxTimeline::SetSpeed(int iSpeed)
+{
+  for (auto &demuxer : m_demuxer)
+    demuxer->SetSpeed(iSpeed);
+}
+
+int CDemuxTimeline::GetStreamLength()
+{
+  return m_chapterMap.rbegin()->first;
+}
+
+std::vector<CDemuxStream*> CDemuxTimeline::GetStreams() const
+{
+  return m_primaryDemuxer->GetStreams();
+}
+
+int CDemuxTimeline::GetNrOfStreams() const
+{
+  return m_primaryDemuxer->GetNrOfStreams();
+}
+
+std::string CDemuxTimeline::GetFileName()
+{
+  return m_primaryDemuxer->GetFileName();
+}
+
+void CDemuxTimeline::EnableStream(int id, bool enable)
+{
+  for (auto &demuxer : m_demuxer)
+    demuxer->EnableStream(demuxer->GetDemuxerId(), id, enable);
+}
+
+CDemuxStream* CDemuxTimeline::GetStream(int iStreamId) const
+{
+  return m_primaryDemuxer->GetStream(m_primaryDemuxer->GetDemuxerId(), iStreamId);
+}
+
+std::string CDemuxTimeline::GetStreamCodecName(int iStreamId)
+{
+  return m_primaryDemuxer->GetStreamCodecName(m_primaryDemuxer->GetDemuxerId(), iStreamId);
+}
+
+
+CDemuxTimeline* CDemuxTimeline::CreateTimeline(CDVDDemux *demuxer)
+{
+  return CreateTimelineFromMatroskaParser(demuxer);
+  return CreateTimelineFromEbml(demuxer);
+}
+
+std::string segUidToHex(std::string uid)
+{
+	const char *hex = "0123456789abcdef";
+	std::string result("0x");
+	result.reserve(18);
+	for (unsigned char twoDigits : uid)
+	{
+		result.append(1, hex[(twoDigits >> 4) & 0xf]);
+		result.append(1, hex[twoDigits & 0xf]);
+	}
+	return result;
+}
+
+CDemuxTimeline* CDemuxTimeline::CreateTimelineFromMatroskaParser(CDVDDemux *primaryDemuxer)
+{
+  std::unique_ptr<CDVDInputStreamFile> inStream(new CDVDInputStreamFile(CFileItem(primaryDemuxer->GetFileName(), false)));
+  if (!inStream->Open())
+    return nullptr;
+  CDVDInputStream *input = inStream.get();
+
+  MatroskaFile mkv;
+  bool result = mkv.Parse(input);
+  if (!result)
+    return nullptr;
+
+   // multiple editions unsupported (at the moment)
+  if (mkv.segment.chapters.editions.size() != 1)
+    return nullptr;
+  auto &edition = mkv.segment.chapters.editions.front();
+  // only handle ordered chapters
+  if (!edition.flagOrdered)
+    return nullptr;
+
+  std::unique_ptr<CDemuxTimeline> timeline(new CDemuxTimeline);
+  timeline->m_primaryDemuxer = primaryDemuxer;
+  timeline->m_demuxer.emplace_back(primaryDemuxer);
+
+  // collect needed segment uids
+  std::set<MatroskaSegmentUID> neededSegmentUIDs;
+  for (auto &chapter : edition.chapterAtoms)
+    if (chapter.segUid.size() != 0 && chapter.segUid != mkv.segment.infos.uid)
+      neededSegmentUIDs.insert(chapter.segUid);
+
+  // find linked segments
+  std::map<MatroskaSegmentUID,CDVDDemux*> segmentDemuxer;
+  segmentDemuxer[""] = primaryDemuxer;
+  segmentDemuxer[mkv.segment.infos.uid] = primaryDemuxer;
+  std::list<std::string> searchDirs({""}); // should be a global setting
```

Comment:
yes. in my case i want the "common_sub_dirs" to be a global setting, so i would need to add it there.
I am not sure about using Util::GetItemsToScan, like it is done in ScanForExternalAudio, because it doesn't search recursiv for subdirs and so something like "OPandED/OP/" would not be found.